browser/net/brave_ad_block_tp_network_delegate_helper.cc

```diff
@@ -194,10 +196,9 @@ void OnBeforeURLRequestAdBlockTP(const ResponseCallback& next_callback,
   scoped_refptr<base::SequencedTaskRunner> task_runner =
       g_brave_browser_process->ad_block_service()->GetTaskRunner();
 
-  DCHECK(ctx->browser_context);
   // DoH or standard DNS quries won't be routed through Tor, so we need to skip
   // it.
-  if (ctx->browser_context->IsTor()) {
+  if (!ctx->browser_context || ctx->browser_context->IsTor()) {
```

Comment:
If we can mock out a `BrowserContext` or move this to a field of `BraveRequestInfo`, then we can properly unit-test CNAME uncloaking as well.